### PR TITLE
docs: Add GraphQL docs for unset field

### PR DESCRIPTION
**Pull request by Moumouls**

### CODE REVIEW

**Comment by mtrezza** [comment]
We don't have versioned docs yet, so this has to sit until 5.0 release

**Comment by Moumouls** [comment]
@mtrezza  this one should also be ok to merge 🙂

File: _includes/graphql/files.md [hunk 1]
```diff
@@ -179,3 +179,38 @@ mutation createGameScore($file: Upload! ) {
   }
 }
 ```
+
+## Unlink a file
+
+Lets update a `GameScore` object and unset the file linked on the `screenshot` field. By setting the `screenshot` field to null, the linked file will be removed from the `Gamescore` object.
```
Comment (mtrezza) [inline _includes/graphql/files.md]: > Let's

> linked in the `screenshot` field

File: _includes/graphql/files.md [hunk 2]
```diff
@@ -179,3 +179,38 @@ mutation createGameScore($file: Upload! ) {
...
+
+**Note:** the file will be not deleted from your file storage.
```
Comment (mtrezza) [inline _includes/graphql/files.md]: > The file

File: _includes/graphql/objects.md [hunk 3]
```diff
@@ -116,6 +116,50 @@ mutation updateAGameScore {
 ```
 **Note:** If you use [Apollo Client](https://www.apollographql.com/docs/react/) it's recommended to request the modified fields and `id` during the Mutation, then the [Apollo Client](https://www.apollographql.com/docs/react/) will automatically update its local store and push the new data across your app; i.e. If you update `playerName` you should request `playerName` and `id` like the code above.
 
+### Unset a field
+Across the whole GraphQL API you can simply unset a field by setting his value to `null`.
+
+**Note:** Following the official GraphQL API Specs, setting a field to `null` through the GraphQL API will completly unset the field in the database on the targeted Parse Object. GraphQL API will transform `null` on the server before saving the object to correctly unset the field into the database.
```
Comment (mtrezza) [inline _includes/graphql/objects.md]: > Following the official GraphQL API specification, setting a field to `null` will unset the field in the database for the  respective Parse Object. The GraphQL API will transform `null` on the server side before saving the object to correctly unset the field in the database.

Remove `Note:` before the comment, doesn't seem to be necessary.

File: _includes/graphql/objects.md [hunk 4]
```diff
@@ -116,6 +116,50 @@ mutation updateAGameScore {
...
+      "gameScore": {
+        "id": "R2FtZVNjb3JlOmM3TVpDZEhQY2w=",
+        "playerName": null
+      }
+    }
+  }
+}
+```
+
+**Additional note:** GraphQL API will always return `null` if the field is `null` or `undefined` in the database. A GraphQL API do not support difference between `null` and `undefined` into the data response.
```
Comment (mtrezza) [inline _includes/graphql/objects.md]: > The GraphQL API will return `null` if the field is `null` or `undefined` in the database. The GraphQL API do not differentiate between `null` and `undefined` in the data response.

Remove `Additional note:` before the comment, doesn't seem to be necessary.

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by Moumouls** [comment]
@mtrezza PR updated, also my VS code has performed a quick lint on the MD.

File: _includes/graphql/objects.md [hunk 5]
```diff
@@ -114,8 +111,53 @@ mutation updateAGameScore {
   }
 }
 ```
+
 **Note:** If you use [Apollo Client](https://www.apollographql.com/docs/react/) it's recommended to request the modified fields and `id` during the Mutation, then the [Apollo Client](https://www.apollographql.com/docs/react/) will automatically update its local store and push the new data across your app; i.e. If you update `playerName` you should request `playerName` and `id` like the code above.
 
+### Unset a field
+
+Across the whole GraphQL API you can simply unset a field by setting his value to `null`.
```
Comment (mtrezza) [inline _includes/graphql/objects.md]:
```suggestion
Across the whole GraphQL API you can simply unset a field by setting its value to `null`.
```

File: _includes/graphql/files.md [hunk 6]
```diff
@@ -179,3 +176,36 @@ mutation createGameScore($file: Upload! ) {
   }
 }
 ```
+
+## Unlink a file
+
+Let's update a `GameScore` object and unset the file linked in the `screenshot` field. By setting the `screenshot` field to null, the linked file will be removed from the `Gamescore` object.
```
Comment (mtrezza) [inline _includes/graphql/files.md]:
```suggestion
Let's update a `GameScore` object and unset the file linked in the `screenshot` field. By setting the `screenshot` field to `null`, the linked file will be removed from the `Gamescore` object.
```

File: _includes/graphql/objects.md [hunk 7]
```diff
@@ -114,8 +111,53 @@ mutation updateAGameScore {
...
+      "gameScore": {
+        "id": "R2FtZVNjb3JlOmM3TVpDZEhQY2w=",
+        "playerName": null
+      }
+    }
+  }
+}
+```
+
+GraphQL API will always return `null` if the field is `null` or `undefined` in the database. A GraphQL API do not support difference between `null` and `undefined` into the data response.
```
Comment (mtrezza) [inline _includes/graphql/objects.md]:
```suggestion
The GraphQL API will always return `null` if the field is `null` or `undefined` in the database. The GraphQL API does not differentiate between `null` and `undefined` in the data response.
```

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by Moumouls** [comment]
thanks @mtrezza  for suggested commits tell if we need to adjust other things here 🚀

**Review by mtrezza — APPROVED** [review]
Looks all good, thanks for the fixes, is this ready for merge?

**Comment by Moumouls** [comment]
@mtrezza ready to merge here :)

**Pull request merged**